tools/fp20compiler/rc1.0_general.cpp

```diff
@@ -234,7 +234,7 @@ void GeneralFunctionStruct::Validate(int stage, int portion)
 static void GenerateInput(int portion, char variable, MappedRegisterStruct reg) {
     const char* portion_s = portion == RCP_RGB ? "COLOR" : "ALPHA";
     printf("MASK(NV097_SET_COMBINER_%s_ICW_%c_SOURCE, 0x%x)", portion_s, variable, reg.reg.bits.name);
-    bool alpha_channel = ((portion == RCP_RGB && reg.reg.bits.channel == RCP_BLUE) || (portion == RCP_ALPHA));
+    bool alpha_channel = ((portion == RCP_RGB && reg.reg.bits.channel == RCP_BLUE) || (reg.reg.bits.channel == RCP_ALPHA));

```

Comment:
Reviewing this now, this still looks weird? I don't think using `.bbb` in the RGB portion is even allowed.

I think something like this would make more sense:
```c
// portion is either RGB or ALPHA
// reg.reg.bits.channel appears to be either RGB, ALPHA or BLUE (judging by MAP_CHANNEL)
bool alpha_channel = (reg.reg.bits.channel == RCP_ALPHA);

// For RGB we can select .rgb or .aaa
assert((portion != RCP_RGB) || ((reg.reg.bits.channel == RCP_RGB) || (reg.reg.bits.channel == RCP_ALPHA)));
// For ALPHA we can select .b or .a
assert((portion != RCP_ALPHA) || ((reg.reg.bits.channel == RCP_BLUE) || (reg.reg.bits.channel == RCP_ALPHA)));
```

This would be similar to the existing `MAP_CHANNEL` macro which also only depends on `reg.reg.bits.channel`